actionlist/app/reducers/action_items.js

```diff
@@ -1,13 +1,93 @@
 /* eslint arrow-body-style: ["error", "as-needed", { "requireReturnForObjectLiteral": true }] */
 import { handleActions } from 'redux-actions';
-import { TOGGLE_DRAWER } from '../actions/types';
+import { TOGGLE_DRAWER, SORT_BY_CREATION_DATE, SORT_BY_LAST_APPROVED_DATE, SORT_BY_PROCESS_TYPE, SORT_BY_ACTION_REQUIRED } from '../actions/types';
 import { processInstances } from '../lib/commons';
 
 export const defaultState = {
   dataSource: processInstances,
   drawerExpanded: false,
+  sortValue: '',
 };
 
+const sortByCreationDate = (state) => {
```

Comment:
I'd recommend not manipulating data you get from outside sources in the reducer, but it's not a blocker if you really want to do it this way for sorting. In my experience, you're typically going to need the full, unaltered set of data from the server in *some* component (particularly when you get to filtering because if you filter the data in the reducer, you can't ever get back the items you filtered out), so it's best to give that to your connected components and let them manipulate the data and give that to their children.